torch/testing/_asserts.py

```diff
@@ -154,58 +158,71 @@ def _equalize_attributes(a: torch.Tensor, b: torch.Tensor) -> Tuple[torch.Tensor
     return a, b
 
 
-_Trace = namedtuple("Trace", ("total", "abs", "rel", "idx", "diff", "a", "b"))
+_Trace = namedtuple(
+    "Trace",
+    (
+        "total_elements",
+        "total_mismatches",
+        "mismatch_ratio",
+        "max_abs_diff",
+        "max_abs_diff_idx",
+        "max_rel_diff",
+        "max_rel_diff_idx",
+    ),
+)
 
 
 def _trace_mismatches(a: torch.Tensor, b: torch.Tensor, mismatches: torch.Tensor) -> _Trace:
     """Traces mismatches and returns the found information.
-
     The returned named tuple has the following fields:
-    - total (int): Total number of values in :attr:`a` and :attr:`b`.
-    - abs (int): Absolute number of mismatches.
-    - rel (float): Relative number of mismatches.
-    - idx (Union[int, Tuple[int, ...]]): Index of greatest absolute difference.
-    - diff (Union[int, float]): Greatest absolute difference.
-    - a (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
-    - b (Union[int, float]): Value of :attr:`a` at the greatest absolute difference.
+    - total_elements (int): Total number of values.
+    - total_mismatches (int): Total number of mismatches.
+    - mismatch_ratio (float): Quotient of total mismatches and total elements.
+    - max_abs_diff (Union[int, float]): Greatest absolute difference of :attr:`a` and :attr:`b`.
```

Comment:
This looks really good; we don't need to block this diff in the stack on this question, but is NaN vs 0 a greater absolute difference than inf vs 0? What about 5 vs 0?

Reply:
```python
>>> torch.sort(torch.tensor([5, float("inf"), float("nan")])).values
tensor([5., inf, nan])
```

This means in case any NaN or Inf value is present it will always be used as greatest diff. We should definitively revisit this in the future.